pkg/controller/istiocontrolplane/listeners.go

```diff
@@ -0,0 +1,264 @@
+	switch chartName {
+	case "istio/charts/kiali":
+		return NewKialiChartCustomizer(chartName, f.DefaultChartCustomizerFactory.ChartAnnotationKey)
+	default:
+		return NewIstioDefaultChartCustomizer(chartName, f.DefaultChartCustomizerFactory.ChartAnnotationKey)
+	}
+}
+
+// IstioDefaultChartCustomizer represents the default ChartCustomizer for IstioControlPlane charts.
+type IstioDefaultChartCustomizer struct {
+	*helmreconciler.DefaultChartCustomizer
+}
+
+var _ helmreconciler.ChartCustomizer = &IstioDefaultChartCustomizer{}
+
+// NewIstioDefaultChartCustomizer creates a new IstioDefaultChartCustomizer
+func NewIstioDefaultChartCustomizer(chartName, chartAnnotationKey string) *IstioDefaultChartCustomizer {
+	return &IstioDefaultChartCustomizer{
+		DefaultChartCustomizer: helmreconciler.NewDefaultChartCustomizer(chartName, chartAnnotationKey),
+	}
+}
+
+// EndChart waits for any deployments or stateful sets that were created to become ready
+func (c *IstioDefaultChartCustomizer) EndChart(chartName string) error {
+	// ignore any errors.  things should settle out
+	c.waitForDeployments()
+	return nil
+}
+
+func (c *IstioDefaultChartCustomizer) waitForDeployments() {
+	if statefulSets, ok := c.NewResourcesByKind["StatefulSet"]; ok {
+		for _, statefulSet := range statefulSets {
+			c.waitForDeployment(statefulSet)
+		}
+	}
+	if deployments, ok := c.NewResourcesByKind["Deployment"]; ok {
+		for _, deployment := range deployments {
+			c.waitForDeployment(deployment)
+		}
+	}
+}
+
+// XXX: configure wait period
+func (c *IstioDefaultChartCustomizer) waitForDeployment(object runtime.Object) {
+	gvk := object.GetObjectKind().GroupVersionKind()
+	logger := c.Reconciler.GetLogger()
+	objectAccessor, err := meta.Accessor(object)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("could not get object accessor for %s", gvk.Kind))
+		return
+	}
+	name := objectAccessor.GetName()
+	namespace := objectAccessor.GetNamespace()
+	deployment := &unstructured.Unstructured{}
+	deployment.SetGroupVersionKind(gvk)
+	// wait for deployment replicas >= 1
+	logger.Info("waiting for deployment to become ready", gvk.Kind, name)
+	err = wait.ExponentialBackoff(wait.Backoff{Duration: 6 * time.Second, Steps: 10, Factor: 1.1}, func() (bool, error) {
+		err := c.Reconciler.GetClient().Get(context.TODO(), client.ObjectKey{Namespace: namespace, Name: name}, deployment)
+		if err == nil {
+			val, _, _ := unstructured.NestedInt64(deployment.UnstructuredContent(), "status", "readyReplicas")
+			return val > 0, nil
+		} else if errors.IsNotFound(err) {
+			logger.Error(nil, "attempting to wait on unknown deployment", gvk.Kind, name)
+			return true, nil
+		}
+		logger.Error(err, "unexpected error occurred waiting for deployment to become ready", gvk.Kind, name)
+		return false, err
+	})
+	if err != nil {
+		logger.Error(nil, "deployment failed to become ready in a timely manner", gvk.Kind, name)
+	}
+}
+
+// KialiChartCustomizer is a ChartCustomizer for the kiali chart
```

Comment:
This should probably be in a separate file.

Reply:
yeah, they can be split up, but right now, there's not really any customization besides the status field.  these wait functions could be generic.  i don't have an opinion one way or the other.

Reply:
this can be done in a future pr.